packages/bundler/src/Bundler.ts

```diff
@@ -186,18 +190,19 @@ export class Bundler implements IBundler {
 
         return new Promise<UserOpStatus>((resolve, reject) => {
           const intervalValue = this.UserOpWaitForTxHashIntervals[chainId] || 500; // default 0.5 seconds
-          const intervalId = setInterval(() => {
-            this.getUserOpStatus(sendUserOperationResponse.result)
-              .then((userOpStatus) => {
-                if (userOpStatus && userOpStatus.state && userOpStatus.transactionHash) {
-                  clearInterval(intervalId);
-                  resolve(userOpStatus);
-                }
-              })
-              .catch((error) => {
+          const intervalId = setInterval(async () => {
```

Comment:
This changes also doesn't make any difference other than using async instead of promise then catch pattern.

Reply:
You're correct on the fact that this is simply changing `.then`/`.catch` to `async`/`await`.
But it does change the behavior, because the timeout check below was not within the `.then` block.
So now, the timeout check will only run after the actual verification of the status of the user op has finished.
This change was made for this function to be consistent with the other `wait` function.

Reply:
Hmm It makes sense that the timeout check is conditioned on the completion of the getUserOpStatus operation, making the behavior more consistent with the outcome of that operation. 

But also in this case, imo it should not be a problem even if it would run independently of the resolution or rejection of the promise returned by getUserOpStatus.

However it improves readability and makes consistent